pkg/task/meta.go

```diff
@@ -26,137 +26,210 @@ import (
-	PatchTA TaskAction = "patch"
+	PatchTA MetaTaskAction = "patch"
 	// OutputTA flags the task action as output. Typically used to
 	// provide a schema (i.e. a custom defined) based output after
 	// running one or more tasks.
-	OutputTA TaskAction = "output"
+	OutputTA MetaTaskAction = "output"
 )
 
-// MetaTask contains information about a Task
-type MetaTask struct {
-	// TaskIdentity provides the required identity
-	// about the task
-	TaskIdentity
+// MetaTaskProps provides properties representing the task's meta
+// information
+type MetaTaskProps struct {
 	// RunNamespace is the namespace where task will get
 	// executed
 	RunNamespace string `json:"runNamespace"`
-	// Action to be invoked on the task
-	Action TaskAction `json:"action"`
 	// Owner represents the owner of this task
 	Owner string `json:"owner"`
-	// ObjectName is the name of the target that is
-	// created or operated by the task
+	// ObjectName is the name of the resource that gets
+	// created or will get operated by this task
 	ObjectName string `json:"objectName"`
 	// Options is a set of selectors that can be used for
-	// get or list actions
+	// tasks that are get or list based actions
 	Options string `json:"options"`
-	// @Deprecated
-	//  In favour of 'RunTask.post' property
-	//
-	// TaskResultQueries will consist of the queries to be run against the
-	// task's result
-	TaskResultQueries []TaskResultQuery `json:"queries"`
-	// Retry specifies the no. of times this task can be tried
-	// This is typically used along with task result verify options
-	// for get or list related actions
+	// Retry specifies the no. of times this same task (i.e. all proeprties
+	// remains same) can be re-tried. This is typically used along with task
+	// result verify options for get or list related actions
 	//
 	// A sample retry option:
 	//
 	// # max of 10 attempts in 20 seconds interval
 	// retry: "10,20s"
 	Retry string `json:"retry"`
-	// RepeatWith sets one or more resources of a given kind that leads to
-	// repetitive execution of a task (in other words a task template is executed
-	// multiple times). The repetition count is determined by the resource names
-	// specified. Each task execution is based on a resource from this provided
-	// list of resources.
+}
+
+// toString returns a string representation of MetaTaskProps structure. In this
+// representation, each property is separated from its value via '='. In
+// addition each property=value pair is separated from other pair via '::'.
+//
+// Example:
+//  runNamespace=default::objectName=MySvc::retry=3,20s
+func (m MetaTaskProps) toString() string {
+	return fmt.Sprintf("runNamespace=%s::owner=%s::objectName=%s::options=%s::retry=%s",
+		m.RunNamespace,
+		m.Owner,
+		m.ObjectName,
+		m.Options,
+		m.Retry)
+}
+
+// selectOverride will override the current meta task properties from the given
+// if the given's properties has value
+func (m MetaTaskProps) selectOverride(given MetaTaskProps) MetaTaskProps {
```

Comment:
Can this be a pointer receiver as we are after all mutating the object?

Reply:
Maybe. I am not sure of performance implications here (if performance is the concern for this review comment). This is a small struct which is embedded inside metaTaskExecutor.

I would park these for future refactoring as well. IMO this struct along with others which are currently being embedded inside metaTaskExecutor might actually converted as `interfaces` and these interfaces can be composed inside metaTaskExecutor instead. In other words, single method interfaces should help in making the code clean and decoupled. The receiver can be pointer implementing these interfaces.

Reply:
:+1: